mlflow/fastai.py

```diff
@@ -157,7 +158,10 @@ def __init__(self, learner):
     def predict(self, dataframe):
         test_data = TabularList.from_df(dataframe, cont_names=self.learner.data.cont_names)
         self.learner.data.add_test(test_data)
-        return self.learner.get_preds(DatasetType.Test)[0].numpy()
+        preds = self.learner.get_preds(DatasetType.Test)
+        preds = [pred.numpy() for pred in preds]
+        preds = [pred if len(pred.shape) > 1 else pred.reshape((len(pred), 1)) for pred in preds]
+        return np.concatenate(preds, axis=1)
```

Comment:
@antoniomdk thanks for the update! I was actually thinking here that we'd return a pandas DataFrame with explicit `prediction` and `target` columns, corresponding to the first & second elements of the tuple returned by `self.learner.get_preds`, respectively. I don't think we'd want to concatenate the predictions and targets together into a numpy single array, as this makes it hard for the user of the model to interpret the result (they need to know how many output classes the model produces & write logic to parse out the predictions/targets separately). 

For example, given a binary classification model with:

```
>>> preds, targets = self.learner.get_preds(DatasetType.Test)
>>> print(preds)
[[0.1 0.9]
 [0.9 0.1]]
>>> preds.shape
(2, 2)
>>> print(targets)
[0 1]
>>> targets.shape
(2,)
```

We'd want to return a Pandas dataframe where:
* The `prediction` column contains 1D arrays with class probabilities, e.g. in the example above the dataframe would contain two rows - one row with `prediction` value `[0.1, 0.9]` and one row with value `[0.9, 0.1]`. 
* The `target` column contains scalar target values, e.g. in the example above the target column would contain values `[0, 1]`

Note that packaging the result in this way is actually important in order for it to work properly with MLflow's Spark UDF functionality ([link](https://github.com/mlflow/mlflow/blob/402381702e8c4b8383a1910062dc1907761c05ac/mlflow/pyfunc/__init__.py#L453-L455)), which will attempt to convert the result of `predict` directly into a Pandas DataFrame if it isn't already one. In the current implementation, this would result in:

```
>>> concatenated = np.concatenate([preds, targets.reshape(len(targets), 1)], axis=1)
>>> concatenated
array([[0.1, 0.9, 0. ],
       [0.9, 0.1, 1. ]])
>>> pd.DataFrame(data=concatenated)
     0    1    2
0  0.1  0.9  0.0
1  0.9  0.1  1.0
```

Thus the current implementation would cause the Spark UDF to return a DataFrame with three columns for the above example, rather than one array-valued column containing class probabilities & another scalar column containing the actual targets.

Reply:
Ah okay, I got it wrong.  Thanks for the clarification.  I wasn't sure how it would work if the wrapper returns a non-numerical column (dtype=object). I will change that, I just hope it won't cause any serialization issue.